pkg/sdk/alert.go

```diff
@@ -0,0 +1,421 @@
+		return errors.New(`
+		Only one of the following actions can be performed at a time:
+		{
+			RESUME | SUSPEND,
+			SET,
+			UNSET,
+			MODIFY CONDITION EXISTS,
+			MODIFY ACTION
+		}
+		`)
+	}
+
+	return nil
+}
+
+type AlertSet struct {
+	Warehouse *AccountObjectIdentifier `ddl:"identifier,equals" sql:"WAREHOUSE"`
+	Schedule  *string                  `ddl:"parameter,single_quotes" sql:"SCHEDULE"`
+	Comment   *string                  `ddl:"parameter,single_quotes" sql:"COMMENT"`
+}
+
+type AlertUnset struct {
+	warehouse *bool `ddl:"keyword" sql:"WAREHOUSE"`
+	schedule  *bool `ddl:"keyword" sql:"SCHEDULE"`
+	comment   *bool `ddl:"keyword" sql:"COMMENT"`
+}
+
+func (v *alerts) Alter(ctx context.Context, id SchemaObjectIdentifier, opts *AlterAlertOptions) error {
+	if opts == nil {
+		return errors.New("alter alert options cannot be empty")
+	}
+
+	opts.name = id
+	if err := opts.validate(); err != nil {
+		return err
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	return err
+}
+
+type DropAlertOptions struct {
+	drop  bool                   `ddl:"static" sql:"DROP"`  //lint:ignore U1000 This is used in the ddl tag
+	alert bool                   `ddl:"static" sql:"ALERT"` //lint:ignore U1000 This is used in the ddl tag
+	name  SchemaObjectIdentifier `ddl:"identifier"`
+}
+
+func (opts *DropAlertOptions) validate() error {
+	if !validObjectidentifier(opts.name) {
+		return ErrInvalidObjectIdentifier
+	}
+	return nil
+}
+
+func (v *alerts) Drop(ctx context.Context, id SchemaObjectIdentifier) error {
+	// alert drop does not support [IF EXISTS] so there are no drop options.
+	opts := &DropAlertOptions{
+		name: id,
+	}
+	if err := opts.validate(); err != nil {
+		return fmt.Errorf("validate alert options: %w", err)
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+type ShowAlertOptions struct {
+	show   bool  `ddl:"static" sql:"SHOW"` //lint:ignore U1000 This is used in the ddl tag
+	Terse  *bool `ddl:"keyword" sql:"TERSE"`
+	alerts bool  `ddl:"static" sql:"ALERTS"` //lint:ignore U1000 This is used in the ddl tag
+
+	// optional
+	Like       *Like   `ddl:"keyword" sql:"LIKE"`
+	In         *In     `ddl:"keyword" sql:"IN"`
+	StartsWith *string `ddl:"parameter,no_equals,single_quotes" sql:"STARTS WITH"`
+	Limit      *int    `ddl:"parameter,no_equals" sql:"LIMIT"`
+}
+
+func (v *Alert) ID() SchemaObjectIdentifier {
+	return NewSchemaObjectIdentifier(v.DatabaseName, v.SchemaName, v.Name)
+}
+
+func (v *Alert) ObjectType() ObjectType {
+	return ObjectTypeAlert
+}
+
+type Alert struct {
+	CreatedOn    time.Time
+	Name         string
+	DatabaseName string
+	SchemaName   string
+	Owner        string
+	Comment      *string
+	Warehouse    string
+	Schedule     AlertSchedule
+	State        string
+	Condition    string
+	Action       string
+}
+
+type AlertSchedule interface {
```

Comment:
i think its an interesting idea to have this as a separate type.  I guess the reason i had settled on using strings is because its a string type in the API documentation.

Note that are actually other places which use schedules, for example [ Failover Groups](https://docs.snowflake.com/en/sql-reference/sql/create-failover-group) and [Replication Groups ](https://docs.snowflake.com/en/sql-reference/sql/create-replication-group). In the failover groups file, you can see i just treat it like a string instead of using a separate type. We could have a "type Schedule struct" which could have a String() method to convert it into a schedule. Alternatively i can edit the sql_builder to be able to allow wrapping keywords with quotes. Or we can just keep it as a *string. Let me know your preference. I do not like having a interface though, i think thats a little too different than the pattern we have been going with for the rest of the SDK.

Reply:
Yes, I did not notice other resources

Reply:
Test,
test

Reply:
Test,
test

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for type

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for type

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding some type safety

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding some type safety

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding some

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding some

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I adde

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I used an interface because that's the \"cleaner\"

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I used an interface because that's the \"cleaner\"

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I used an interface because that's the cleanest way of encoding sum types in Go that I know of, but I think a struct will do.

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I used an interface because that's the cleanest way of encoding sum types in Go that I know of, but I think a struct will do.

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I used an interface because that's the cleanest way of encoding sum types in Go that I know of, but I think a struct will do.
side note: In `Create`

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I used an interface because that's the cleanest way of encoding sum types in Go that I know of, but I think a struct will do.
side note: In `Create`

Reply:
Fir

Reply:
Fir

Reply:
Test comment

Reply:
Test comment

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I added

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I added

Reply:
abc

Reply:
abc

Reply:
Test comment
Test commentt

Reply:
Test comment
Test commentt

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I used an interface because that's the cleanest way of encoding sum types in Golang that I know of, but a struct will do too.
side note: In `Create` I can use `AlertSchedule` type, but in `CreateAlertOptions` I'm forced to use string.
What do you think about updating sql_builder so it also can handle types implementing `Stringer` interface?

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I used an interface because that's the cleanest way of encoding sum types in Golang that I know of, but a struct will do too.
side note: In `Create` I can use `AlertSchedule` type, but in `CreateAlertOptions` I'm forced to use string.
What do you think about updating sql_builder so it also can handle types implementing `Stringer` interface?

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I used an interface because that's the cleanest way of encoding sum types in Golang that I know of, but a struct will do too.
side note: In `Create` I can use `AlertSchedule` type, but in `CreateAlertOptions` I'm forced to use string.
What do you think about updating sql_builder so it also can handle types implementing `Stringer` interface?
If `AlertSchedule` was not required, it would not be a separate argument in the function, so even in the interface we would 
just have a plain string.

Reply:
Yes, I did not notice it's used by other resources
In general, I'm all for adding more types.
I used an interface because that's the cleanest way of encoding sum types in Golang that I know of, but a struct will do too.
side note: In `Create` I can use `AlertSchedule` type, but in `CreateAlertOptions` I'm forced to use string.
What do you think about updating sql_builder so it also can handle types implementing `Stringer` interface?
If `AlertSchedule` was not required, it would not be a separate argument in the function, so even in the interface we would 
just have a plain string.